### PR TITLE
Add StatsDCollector.close and some docs

### PR DESCRIPTION
**Pull request by dave-shawley**
This PR adds a close method so it is possible to shut down the client cleanly.  I added documentation for the connector, fixed a minor defect (c4dd176), updated development tool versions, and got test coverage up to 100%.

### CODE REVIEW

File: tests.py [hunk 1]
```diff
@@ -48,7 +47,7 @@ def get_app(self):
     def test_bad_protocol_raises_ValueError(self):
         with self.assertRaises(ValueError):
             statsd.StatsDCollector(host='127.0.0.1',
-                                   port=8125,
+                                   port='8125',
```
Comment (andrewrabert) [inline tests.py:50]: Why was this changed?

**Review by andrewrabert — APPROVED** [review]
Just one question.

File: tests.py (hunk 1 above)
Comment (dave-shawley) [reply tests.py:50]: Cuz port is supposed to be a string ;)
https://github.com/sprockets/sprockets.mixins.metrics/blob/0b17d5f0c09a2be9db779e17e6789d3d5ff9a0d0/sprockets/mixins/metrics/statsd.py#L95
